more various l10n updates (13)

tijuca opened the pull request:
Mainly an update due the new added vaults feature. Also same usual tiny improvements to existing strings.

tijuca committed:
l10n: modify log result string concatenation

l10n strings with PHP variables inside wont work as expected, split of
the existings string into more peaces.

tijuca committed:
l10n: make items in vaults action menu translatable

The various elements of the action menu for the various vault types
should be translatable as other actions menus are already be.

tijuca committed:
l10n: more split of result string concatenation

The l10n string translation isn't working in case it includes PHP
variables, extend the concatenation into more parts.

tijuca committed:
l10n: add h4 element to l10n strings

tijuca committed:
l10n: Update German translations

Update the po file and also the binary message catalogue for German after
reworking various source files.

GaryAllan commented on app/upgrade/upgrade-execute.php:
```diff
@@ -24,7 +24,7 @@
 if($Upgrade->upgrade_database()===true) {
 	# print success
-	$Result->show("success", _("Database upgraded successfully! <a class='btn btn-sm btn-default' href='".create_link('dashboard')."'>Dashboard</a>"), false);
+	$Result->show("success", _("Database upgraded successfully!")." <a class='btn btn-sm btn-default' href='".create_link('dashboard')."'>"._("Dashboard")."</a>"), false);
```
php syntax error. Too many closing brackets ")" in line.

```php
."'>"._("Dashboard")."</a>", false);
```

tijuca replied:
Hmm, why I haven't seen this by myself? Fixed by latest force push.

tijuca committed:
l10n: partially revert discoveryCheck.php

The reverted modification was introduced with #3209, moving array field
back so it's not translatable.
